packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -18,10 +19,15 @@ const { handleReferences, handleWebhookUpdate } = require(`./utils`)
 const agent = {
   http: new HttpAgent(),
   https: new HttpsAgent(),
+  http2: new http2wrapper.Agent(),
 }
 
 async function worker([url, options]) {
-  return got(url, { agent, ...options })
+  return got(url, {
```

Comment:
You also need to pass `request: http2wrapper.auto` here, as Got v11 doesn't use the newest version of `http2-wrapper`. Got v12 is going to be released in the next few weeks.

Reply:
Thanks for waying in @szmarczak !

Reply:
oh haha thanks! 

![1_sXntGz_sVG3164fdY03-qQ](https://user-images.githubusercontent.com/71047/122966237-edbd3d00-d33d-11eb-978b-5a565c406b15.jpeg)

Reply:
This should fix it https://github.com/gatsbyjs/gatsby/pull/32012/commits/5fbf61810472abf85c7a0c0142b72478957bdfa8

Thanks btw for all your great work on Got! It's a fantastic piece of software.

Reply:
Random aside while we have you, I profiled this with http/2 & it looks like `normalizeArguments` is using more than its fair share of CPU?

<img width="1230" alt="Screen Shot 2021-06-22 at 9 42 43 AM" src="https://user-images.githubusercontent.com/71047/122966521-4987c600-d33e-11eb-9c2c-5d88726f7ce9.png">

Reply:
> Thanks btw for all your great work on Got! It's a fantastic piece of software.

No problem :) I really enjoy improving it.

> oh haha thanks!

https://github.com/sindresorhus/got/blob/f896aa52abc41fe40d4942da94a0408477358f14/source/core/index.ts#L2364-L2367

This hasn't been documented, sorry. One of the ways to fix this in Got would be to always pass the entire agent object to the `request` function, but then we would need to default `request` to something like this:

```js
request = (url, options) => {
	if (url.protocol === 'https:') {
		options.agent = options.agent.https;
		return https.request(url, options);
	}

	options.agent = options.agent.http;
	return http.request(url, options);
};
```

But then `request: https.request` wouldn't work. We could detect native `http` functions but then it would fail on custom functions that at the end return the same what `https.request` does.

I'm not sure what's the best solution here. Alternatively we could make `request` an object with protocols `http2` `https` and `http` as keys...

For now `http2: true` does the trick :P

> It looks like normalizeArguments is using more than its fair share of CPU?

Indeed. Can you point it down to a few lines? I suppose it's caused by too intensive object creation. The upcoming Got version uses [getters & setters](https://github.com/sindresorhus/got/blob/a99262c32b4f7513ee5cbdb4f20824d342a18eff/source/core/options.ts#L756) in order to avoid re-doing normalization if not necessary. If you could create a reproducible repo, that would be awesome.

Also feel free to create an issue in the Got repo about this :)